### ISSUE TITLE
Theme Roller for CSS Framework

### ISSUE THREAD

**Issue by sfrisk** [jquery/css-chassis#20]
This is based on the conversation in #17. We should have a way for users to modify variables in a variable file to help customize their download of the css framework to best match their design vision.

**Comment by prasunanand** [comment]
Looking at the themeroller website of jquery UI. I don't think its very user friendly. I would suggest of  adding guiders that allow the user to easily define the components of their theme in a step-wise manner. In the chassis 1.0 , I don't think we would be having a lot of components but still we should be thinking of a proper UX that makes the devs adopt Chassis to create their custom theme as it develops.

**Comment by sfrisk** [comment]
We're looking to have a theme roller closer aligned with [jQuery Mobile's themeroller](http://themeroller.jquerymobile.com/), rather than jQuery UI's

**Comment by prasunanand** [comment]
For the swatches we should be taking the components and not the the entire website as in Jquery Mobile. And my stress is on creating guiders that allows devs to define their theme in a stepwise manner.

**Comment by geekman-rohit** [comment]
The jQuery mobile theme roller is pretty user friendly, I like the idea of multiple swatches certainly something we should consider in the long run.
I have one query @sfrisk , the themeroller will generate onlly a variables.scss file? or compiled css?

Both will have their benefits, we should start with variables.scss file, and later add an additional option to output compiled css if the user wants. Not all of our users will know sass to compile it themselves?

**Comment by arschmitz** [comment]
@prasunanand swatches are a concept that wont exist in chassis this is something specific to jquery mobile and is going to be going away. It basically provided a way to create multiple themes within a single theme. We decided it turned out to be a bad idea in the long run. More just look at the ui and how the themeroller works.

**Comment by arschmitz** [comment]
@geekman-rohit it will create both.

**Comment by geekman-rohit** [comment]
In that case, @arschmitz , would it be a good idea to have the custom builder(allowing users to choose which parts they want : form,table,grid etc.) in the theme-roller itself? Or will that be separate?

**Comment by arschmitz** [comment]
I don't think we had fully decided this yet so far we have been discussing them as different things. However its possible for them to be a single thing.

**Comment by geekman-rohit** [comment]
Here's why I like the idea: 
While a separate online download builder simplifies things, it would be great for users to remove certain elements from their theme from within theme builder.

For example, a user may need only a certain few elements like say table, but does not need a grid system.  It makes sense to allow the user to remove the grid while we build a theme for him?

We can have advanced options inside the theme builder, where user can remove certain parts if required.

**Comment by arschmitz** [comment]
@geekman-rohit another option to handle this is create a link for the theme, and paste that into the builder. jQuery ui took this approach in their builder / themeroller.

**Comment by geekman-rohit** [comment]
@arschmitz, yes that sounds better actually! That gives us the best of both worlds, simplifies, separates concerns and provides flexibilty.
